### PR TITLE
6122 packages info on for Linux

### PR DESCRIPTION
**Pull request by konangc87**
# Packages Info on sysinfo for Linux

## **Related Issue**
[6122](https://github.com/wazuh/wazuh/issues/6122)

## Description
This issue aims to refactor the packages info provider for each os using C++ reducing the scope and responsibilities of the modules..
These changes have to be made in the various implementations according to the operating system targeted for this feature.

## DoD / Tasks
- [X] Implementation
- [X] Run testtool on Ubuntu
image
- [X] Run testtool on Centos
Centos 8
image
Centos 5
image
- [X] SysInfo UT
image

### CODE REVIEW

**Review by Dwordcito — APPROVED** [review]

File: src/wazuh_modules/syscollector/sysInfoLinux.cpp [hunk 1]
```diff
@@ -11,11 +11,21 @@
...
 #include <iostream>
+#include <sys/types.h>
+#include <sys/stat.h>
 #include "stringHelper.h"
+#include "cmdHelper.h"
 
 constexpr auto WM_SYS_HW_DIR{"/sys/class/dmi/id/board_serial"};
 constexpr auto WM_SYS_CPU_DIR{"/proc/cpuinfo"};
 constexpr auto WM_SYS_MEM_DIR{"/proc/meminfo"};
+#define DPKG_PATH "/var/lib/dpkg/"
```
Comment (tinachoB) [inline src/wazuh_modules/syscollector/sysInfoLinux.cpp:22]: Could you please change this define to a constexpr auto ? I think it'd be better to maintain code consistency

File: src/wazuh_modules/syscollector/sysInfoLinux.cpp [hunk 2]
```diff
@@ -44,6 +55,176 @@ static bool getSystemInfo(const std::string& fileName, const std::string& separa
...
+            const auto package{parseRpm(rawData.substr(pos))};
+            if (!package[0].empty())
+            {
+                ret.push_back(package[0]);
+            }
+            rawData = rawData.substr(0, pos);
+            pos = rawData.rfind("Name");
+        }
+    }
+    else
```
Comment (tinachoB) [inline src/wazuh_modules/syscollector/sysInfoLinux.cpp:222]: Do we need this else? If not, should we add a comment saying why else is not needed or is not needed at all?

**Review by tinachoB — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/syscollector/sysInfoLinux.cpp (hunk 1 above)
Comment (konangc87) [reply src/wazuh_modules/syscollector/sysInfoLinux.cpp:22]: will do

File: src/wazuh_modules/syscollector/sysInfoLinux.cpp (hunk 2 above)
Comment (konangc87) [reply src/wazuh_modules/syscollector/sysInfoLinux.cpp:222]: will do

**Review by tinachoB — APPROVED** [review]

**Pull request merged**